base/ltmarks.dtx

```diff
@@ -1158,7 +1420,7 @@
 %    To retrieve the first, last or top region mark, we grab the
 %    appropriate value stored in the corresponding token list variable
 %    and pass its contents back. These functions should be used only
-%    in output routines after \cs{@@_update_structure:nn} has acted,
+%    in output routines after \cs{@@_update_structure_from_material:nn} has acted,
```

Comment:
??

in an output routine and only after . . . 

OR?? 

. . . only immediately after . . .

Reply:
nothing "immediate" here.  The update_structure_from_material is usually called once in the OR (near the beginning) and then for the whole time tha OR is active you can retrieve the marks, e.g., while attaching headers and footers.